horace_core/sqw/file_io/load_sqw_dnd.m

```diff
@@ -0,0 +1,113 @@
+function [data, unmatched_args] = load_sqw_dnd(sources, varargin)
+%LOAD_SQW_DND attempt to load the given sources into sqw/dnd objects
+%
+% Input:
+% ------
+% sources       Can be:
+%                 - sqw object (scalar, array or cell array)
+%                 - dnd object (scalar, array or cell array)
+%                 - char array (path to an sqw/dnd file)
+%                 - cellstr (cell array of sqw/dnd file paths)
+%               You cannot mix types within cell arrays (e.g. a cell array of
```

Comment:
Is this restriction necessary for the function use-cases or a result of the structure of the return values?
Returning array of object limits you with the current sqw/dnd old-style classes, but the new SQW objects would all fit in an array of SQWDnDBase.

Reply:
Update: SQWDnDBase does not extend the Heterogeneous mixin so it's not possible to create a mixed array of DnD and SQW class objects.
The restriction in the comments will remain valid